Lib/test/test_zipfile/test_core.py

```diff
@@ -1780,6 +1780,34 @@ def test_writestr_extended_local_header_issue1202(self):
                 zinfo.flag_bits |= zipfile._MASK_USE_DATA_DESCRIPTOR  # Include an extended local header.
                 orig_zip.writestr(zinfo, data)
 
+    def test_write_with_source_date_epoch(self):
+        # Set the SOURCE_DATE_EPOCH environment variable to a specific timestamp
+        os.environ['SOURCE_DATE_EPOCH'] = "1727440508"
```

Comment:
This setting may affect other tests. Better would be to use a fixture to set it for the life of this test.

```suggestion
        with os_helper.EnvironmentVarGuard() as env:
            # Set the SOURCE_DATE_EPOCH environment variable to a specific timestamp
            env['SOURCE_DATE_EPOCH'] = "1727440508"
```

(and of course move the tests into that block). Do the same for the unset test.